### DISCUSSION TITLE
Configuring connection with AWS RDS db

### DISCUSSION THREAD

**Discussion by Sreejit7** [prisma/prisma#19187] [Q&A]
I'm new to Prisma and AWS. So apologies if this sounds silly.

 I want to connect to a db hosted in a private VPC in AWS RDS. I have tried connecting using the basic connection URI like so: `postgresql://username:password@dbinstanceid.cgaimxq41enr.ap-south-1.rds.amazonaws.com:5432/dbname` following the [docs](https://www.prisma.io/dataguide/postgresql/setting-up-postgresql-on-rds) but on trying `npx prisma db pull` it shows the `P1001` error saying it can't reach the database server running at the specified path. 

How do I configure the connection in this case? i.e. what sort of extra params and the likes do I need to provide to connect the db with Prisma properly? Please help 🙏🏼 Thanks in advance.

PS: I have searched for similar discussion but there were no specific steps mentioned.

**Comment by nurul3101** [comment]
Hello @Sreejit7 👋 

Are you able to connect to your database through a database IDE with this connection string - `postgresql://username:password@dbinstanceid.cgaimxq41enr.ap-south-1.rds.amazonaws.com:5432/dbname`?

When your RDS instance is in a private VPC, it is not directly accessible from the public internet.

You can setup a Bastion host in your VPC, which is a secure, hardened server through which you can access your private network. Once it's setup, you can create an SSH tunnel from your local machine to the Bastion host, and then connect to your RDS instance. After setting up the SSH tunnel, your local machine would act as if it's part of the VPC, and you should be able to connect to your RDS instance using Prisma with the connection string you've specified.

**Reply by Sreejit7** [reply]
Hey, thanks for your answer @nurul3101. So I understand it'll work locally, connecting the db in the VPC. But how will it work when deployed? 
I also found this reference on accessing a db in a VPC from a client application on [AWS docs](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_VPC.Scenarios.html#USER_VPC.Scenario4)
Do I need to do something like this for it to work?

**Reply by nurul3101** [reply]
It depends on where your application is deployed.

If the application is deployed on an EC2 instance in the same VPC then your application would have direct access to the RDS instance, and you wouldn't need to do anything special to connect to it.  If it is deployed on an EC2 instance in different VPC then you would need to setup VPC peering between the two VPCs. This allows instances in the two VPCs to communicate with each other as if they were in the same network.

**Reply by nurul3101** [reply]
@Sreejit7 Did you had a chance to look at my last response?

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by nurul3101** [reply]
I'll mark this discussion as closed. Please feel free to reopen if needed.

**Reply by dcode0n1-9x** [reply]
1️⃣ Verify Network Connectivity to RDS

Before attempting any database connection, confirm that your machine (or the bastion) can reach the RDS endpoint.
```shell
telnet <RDS_HOST> PORT
``` 

it should give you connected. 
2️⃣ Create an SSH Tunnel via the Bastion Host

```shell
ssh -i "<path_to_the_server.pem>server.pem" -N -L 5433:<RDS_HOST>:5432 ubuntu@<Machine_IP>
```

this would let you create a bridge from the localhost point to that. 
3️⃣ Connect to RDS Using a Local Database URL

```shell
DATABASE_URL=postgresql://<username>:<password>@localhost:5433/<dbname>?sslmode=require
```
